rules/opentitan.bzl

```diff
@@ -232,67 +326,209 @@ def opentitan_binary(
+    keys in `per_device_deps`. The actual artifacts created are an ELF file, a
+    BIN file, the disassembly, and the scrambled (ROM) VMEM file. Each of these
+    output targets performs a bazel transition to the RV32I toolchain to build
+    the target under the correct compiler.
+    Args:
+      @param name: The name of this rule.
+      @param platform: The target platform for the artifacts.
+      @param per_device_deps: The deps for each of the hardware target.
+      @param **kwargs: Arguments to forward to `opentitan_binary`.
+    Emits rules:
+      For each device in per_device_deps entry:
+        cc_binary            named: <name>_<device>
+        obj_transform        named: <name>_<device>_elf
+        obj_transform        named: <name>_<device>_bin
+        elf_to_dissassembly  named: <name>_<device>_dis
+        elf_to_scrambled_rom named: <name>_<device>_scr_vmem
+      filegroup named: <name>
+          with all the generated rules
+    """
+
     deps = kwargs.pop("deps", [])
     targets = []
     for (device, dev_deps) in per_device_deps.items():
         devname = "{}_{}".format(name, device)
-        native.cc_binary(
+
+        # Generate ELF, Binary, and Disassembly
+        targets.extend(opentitan_binary(
             name = devname,
             deps = deps + dev_deps,
-            target_compatible_with = _targets_compatible_with[platform],
-            copts = copts,
-            linkopts = linkopts,
             **kwargs
-        )
-        targets.append(":" + devname + "_elf")
-        obj_transform(
-            name = devname + "_elf",
-            srcs = [devname],
-            format = "elf32-little",
-            suffix = "elf",
+        ))
+
+        # Generate Scrambled ROM VMEM
+        targets.append(":" + devname + "_scr_vmem")
+        elf_to_scrambled_rom(
+            name = devname + "_scr_vmem",
+            srcs = [devname + "_elf"],
             platform = platform,
         )
 
-        if output_bin:
-            targets.append(":" + devname + "_bin")
-            obj_transform(
-                name = devname + "_bin",
-                srcs = [devname],
-                platform = platform,
-            )
-        if output_disassembly:
-            targets.append(":" + devname + "_dis")
-            elf_to_disassembly(
-                name = devname + "_dis",
-                srcs = [devname],
-                platform = platform,
-            )
-        if output_scrambled:
-            targets.append(":" + devname + "_scr")
-            elf_to_scrambled(
-                name = devname + "_scr",
-                srcs = [devname],
-                platform = platform,
-            )
+    native.filegroup(
+        name = name,
+        srcs = targets,
+    )
+
+def opentitan_flash_binary(
+        name,
+        platform = OPENTITAN_PLATFORM,
+        signing_keys = {
+            "test_key_0": "//sw/device/silicon_creator/mask_rom/keys:test_private_key_0",
```

Comment:
I'm not super comfortable with this implicit dependency. I would prefer that it was included literally in every target, or maybe we can give the key a clearer name like `default_test_key`.